torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -204,35 +204,45 @@ at::DimVector inversePermutation(
   }
 }
 
+template <typename T, typename CONTAINER>
+void encode_buf(T value, CONTAINER& buffer) {
+  char* v = reinterpret_cast<char*>(&value);
+  for (int i = 0; i < sizeof(T); i++) {
+    buffer.push_back(*(v++));
+  }
+}
+
 } // namespace
 
 InputsIdLookup::IdLookupReturn InputsIdLookup::lookupId(
     const at::ArrayRef<IValue>& inputs) {
   IdLookupReturn ret;
-  std::stringstream encoded_inputs;
+
+  std::lock_guard<std::mutex> guard(mutex_);
```

Comment:
what is this mutex guarding? we didn't have a mutex here before.

Reply:
`encoding_` is now the buffer that was re-used. Back then we are not reusing containers but doing stringstream, getting rid of the copy gives us ~4us overhead.

Reply:
can you please add a comment linking the mutex with the shared state it guards? (where we define `encoding_` for example, it can be as brief as `// Shared state, guarded by mutex_`